main.tf

```diff
@@ -155,6 +185,38 @@ resource "aws_db_instance" "db_excluding_name" {
   tags = "${merge(var.tags, map("Name", format("%s-%s", var.environment, var.name)), map("Env", var.environment))}"
 }
 
+# The database instance itself with read replica
+
+resource "aws_db_instance" "db_excluding_name_with_read_replica" {
+  count = "${var.database_name == "" && var.engine_type != "aurora" && var.engine_type != "aurora-mysql" && var.engine_type != "aurora-postgresql" && var.replicate_source_db != "" ? 1 : 0}"
+
+  allocated_storage           = "${var.allocated_storage}"
+  allow_major_version_upgrade = "${var.allow_major_version_upgrade}"
+  auto_minor_version_upgrade  = "${var.auto_minor_version_upgrade}"
+  backup_retention_period     = "${var.backup_retention_period}"
+  backup_window               = "${var.backup_window}"
+  copy_tags_to_snapshot       = "${var.copy_tags_to_snapshot}"
+  engine                      = "${var.engine_type}"
+  engine_version              = "${var.engine_version}"
+  final_snapshot_identifier   = "${var.name}"
+  identifier                  = "${var.name}"
+  instance_class              = "${var.instance_class}"
+  license_model               = "${var.license_model}"
+  multi_az                    = "${var.is_multi_az}"
+  parameter_group_name        = "${aws_db_parameter_group.db.id}"
+  password                    = "${var.database_password}"
```

Comment:
Since this needs to be empty anyway, you can just remove it

Reply:
I can't remove that as this is required for a resource

Reply:
Not according to https://www.terraform.io/docs/providers/aws/r/db_instance.html#password. Have you tested without it? If you have then I guess the docs are wrong...